docs/adapter_impl_notes.md

```diff
@@ -1,67 +1,17 @@
-
-For reads, the adapter's job is to interpret the options included in the juttle `read` command into a set of matching email messages, construct json objects representing those messages, and pass them to the Juttle Runtime by calling `emit()`. For writes, the adapter's job is to take the output of programs and "save" that output to the backend by sending emails.
-
-More sophisticated adapters work together with the juttle optimizer to push aggregation operations directly into the backend. For example, to count the number of messages in a given time period you could simply fetch all the messages and have the juttle program perform the counting. However it would be more efficient to count the number of messages via Gmail APIs and simply return the count instead.
+This page talks about the implementation details of the gmail adapter. It can be used with the more general [adapter API guide](https://github.com/juttle/juttle/blob/master/docs/adapters/adapter_api.md) to provide details on how to write a new adapter.
 
 This document describes details on module loading and configuration. There are separate documents that discuss the details of the [read gmail](./read.md) and [write gmail](./write.md) procs.
 
-## The `JuttleAdapterAPI` global
-
-An adapter will need to access functions and objects from the Juttle runtime. All of these functions should be accessed via a global object [JuttleAdapterAPI](https://github.com/juttle/juttle/blob/master/lib/adapters/api.js). [this page](https://github.com/juttle/juttle/blob/master/docs/adapters/adapter_api.md) describes the API object and how to use it in more detail.
-
-## Javascript Modules, Classes and Methods
-
-The Gmail adapter implements a javascript module in [index.js](../index.js). It requires the main module in [lib/index.js](../lib/index.js) via:
-
-```Javascript
-module.exports = require('./lib/');
-```
-
-When the adapter is loaded, the CLI/juttle-engin perform a `require` of the module (i.e. the top-level directory containing `index.js`). `lib/index.js` in turn `require()s` `read.js` and `write.js`, which contain the implementation of the read and write classes, respectively.
-
-The main function exported by `lib/index.js` takes a `config` argument containing the configuration object for the adapter, and returns an object with `name`, `read`, and `write` attributes.  The value for the `name` attribute is `gmail`, corresponding to the `read gmail`/`write gmail` proc in juttle programs. The value for `read` is a class inheriting from `AdapterRead`, which performs the read work of the adapter. The value for `write` is an class inheriting from `AdapterWrite`, which performs the write work of the adapter.
-
-Here's a slighly simplified version of the exported function from `lib/index.js`:
```

Comment:
How about keeping this (with the code snippet below), it seems useful for understanding. Move it to after line 15?